src/OpenTelemetry/DiagnosticSourceInstrumentation/PropertyFetcher.cs

```diff
@@ -76,6 +76,12 @@ public bool TryFetch(object obj, out T value, bool skipObjNullCheck = false)
                 this.innerFetcher = PropertyFetch.Create(obj.GetType().GetTypeInfo(), this.propertyName);
             }
 
+            if (this.innerFetcher == null)
+            {
+                value = default;
+                return false;
+            }
```

Comment:
My quick implementation in #3479 differs a bit from yours, but that's ok. I'm going to close #3479.

Though would you mind adding the following test to your PR? https://github.com/open-telemetry/opentelemetry-dotnet/pull/3479/files#diff-35b3db9f257a0d783ab58f02a27c135395048aa3acd17cc54cbb47436370ede4R70-R79

I think it's good to have a test that just validates the underlying changes to `PropertyFetcher`.